File: frontend/src/types/graph.ts
export interface GraphNode {
  id: string
  label: string
  type: 'note' | 'folder' | 'tag'
  color: string
  size: number
}

export interface GraphEdge {
  source: string
  target: string
  type: string
}

export interface GraphData {
  nodes: GraphNode[]
  edges: GraphEdge[]
}

File: backend/app/schemas/graph.py
from typing import List, Optional
from pydantic import BaseModel

class GraphNode(BaseModel):
    id: str
    label: str
    type: str
    color: str
    size: int

class GraphEdge(BaseModel):
    source: str
    target: str
    type: str

class GraphResponse(BaseModel):
    nodes: List[GraphNode]
    edges: List[GraphEdge]
